### PR TITLE
Update to Rust 2021 edition

### PR DESCRIPTION
**Pull request by maackle**
### TODO:
- [ ] ~~CHANGELOG(s) updated with appropriate info~~
- [ ] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

File: crates/holochain/src/core/ribosome/host_fn/call.rs [hunk 1]
```diff
@@ -17,6 +17,7 @@ pub fn call(
     let results: Vec<Result<ZomeCallResponse, WasmError>> =
         tokio_helper::block_forever_on(async move {
             join_all(inputs.into_iter().map(|input| async {
+                let _ = &input;
```
Comment (freesig) [inline crates/holochain/src/core/ribosome/host_fn/call.rs:20]: This seems bad

**Review by freesig — CHANGES_REQUESTED** [review]

File: crates/holochain/src/core/ribosome/host_fn/call.rs (hunk 1 above)
Comment (maackle) [reply crates/holochain/src/core/ribosome/host_fn/call.rs:20]: @freesig is it this one in particular that seems bad, or all the other instances of this? `cargo fix --edition` added most of these automatically, as per the [rust 2021 migration guide](https://doc.rust-lang.org/edition-guide/rust-2021/disjoint-capture-in-closures.html#migration). This is the only one I added automatically, does this one in particular seem problematic?

Comment (freesig) [reply crates/holochain/src/core/ribosome/host_fn/call.rs:20]: I think the auto fix has failed. `let _ = &input;` makes no sense. cargo fix is not perfect. I think all the "fixes" like this are wrong

Comment (maackle) [reply crates/holochain/src/core/ribosome/host_fn/call.rs:20]: The article describes why this fix is necessary, it's to preserve the variable capturing behavior of closures

**Review by freesig — APPROVED** [review]

**Review by steveeJ — APPROVED** [review]

**Pull request merged**